src/cgame/cg_rocket_draw.cpp

```diff
@@ -1049,35 +1049,24 @@ class UsableBuildableElement : public HudElement
 	{
 		vec3_t        view, point;
 		trace_t       trace;
-		entityState_t *es;
 
 		AngleVectors( cg.refdefViewAngles, view, nullptr, nullptr );
 		VectorMA( cg.refdef.vieworg, ENTITY_USE_RANGE, view, point );
 		CG_Trace( &trace, cg.refdef.vieworg, nullptr, nullptr,
 				  point, cg.predictedPlayerState.clientNum, MASK_SHOT, 0 );
 
-		es = &cg_entities[ trace.entityNum ].currentState;
+		const entityState_t &es = cg_entities[ trace.entityNum ].currentState;
 
-		if ( es->eType == entityType_t::ET_BUILDABLE && BG_Buildable( es->modelindex )->usable &&
-			CG_IsOnMyTeam(*es) )
+		if ( es.eType == entityType_t::ET_BUILDABLE
+				&& es.modelindex == BA_H_ARMOURY
+				&& Distance(cg.predictedPlayerState.origin, es.origin) < ENTITY_USE_RANGE - 0.2f // use an epsilon to account for rounding errors
+				&& CG_IsOnMyTeam(es) )
 		{
-			//hack to prevent showing the usable buildable when you aren't carrying an energy weapon
```

Comment:
to me, this checks is what advertise reloading energy weapons at reactor. I do not like the idea of removing this feature, it's handy

Reply:
idk, I'd say the advertizing is that the reactor is shown with a large ammo stack over it when you are out of ammo. This one is confusing IMO.